core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorMetadata.java

```diff
@@ -266,12 +266,22 @@ default ColumnMetadata getColumnMetadata(ConnectorSession session, ConnectorTabl
 
     /**
      * Get statistics for table for given filtering constraint.
+     *
+     * @deprecated Use {@link #getTableStatistics(ConnectorSession, ConnectorTableHandle)}
```

Comment:
add also `@Deprecated` annotation.

- it will make https://github.com/trinodb/trino/pull/11722 happy 
- it shows up in as a deprecated in an IDE even until you download source/javadoc jars, which isn't done by default

cc @kokosing @hashhar would be good to require that any SPI entity with javadoc tag `@deprecated`  is also annotated as `@Deprecated`. Would that be possible?

Reply:
Thanks! I missed that since the IDE picked up the comment.